dao/src/test/java/org/thingsboard/server/dao/service/BaseTenantServiceTest.java

```diff
@@ -275,4 +323,374 @@ public void testSaveTenantWithIsolatedProfileInMonolithSetup() {
+    }
+
+    private Device createAndSaveDeviceFor(Tenant savedTenant, Customer savedCustomer, DeviceProfile savedDeviceProfile) {
+        Device device = new Device();
+        device.setCustomerId(savedCustomer.getId());
+        device.setTenantId(savedTenant.getId());
+        device.setType("Test type");
+        device.setName("TestType");
+        device.setLabel("Test type");
+        device.setDeviceProfileId(savedDeviceProfile.getId());
+        return deviceService.saveDevice(device);
+    }
+
+    private DeviceProfile createAndSaveDeviceProfileWithProfileDataFor(Tenant savedTenant) {
+        DeviceProfile deviceProfile = new DeviceProfile();
+        deviceProfile.setTenantId(savedTenant.getId());
+        deviceProfile.setTransportType(DeviceTransportType.MQTT);
+        deviceProfile.setName("Test device profile");
+        deviceProfile.setType(DeviceProfileType.DEFAULT);
+        DeviceProfileData profileData = new DeviceProfileData();
+        profileData.setTransportConfiguration(new MqttDeviceProfileTransportConfiguration());
+        deviceProfile.setProfileData(profileData);
+        return deviceProfileService.saveDeviceProfile(deviceProfile);
+    }
+
+    private WidgetsBundle createAndSaveWidgetBundleFor(Tenant savedTenant) {
+        WidgetsBundle widgetsBundle = new WidgetsBundle();
+        widgetsBundle.setTenantId(savedTenant.getId());
+        widgetsBundle.setTitle("Test widgets bundle");
+        widgetsBundle.setAlias("TestWidgetsBundle");
+        widgetsBundle.setDescription("Just a simple widgets bundle");
+        return widgetsBundleService.saveWidgetsBundle(widgetsBundle);
+    }
+
+    private Customer createAndSaveCustomerFor(Tenant savedTenant) {
+        Customer customer = new Customer();
+        customer.setTitle("Test customer");
+        customer.setTenantId(savedTenant.getId());
+        customer.setEmail("testCustomer@test.com");
+        return customerService.saveCustomer(customer);
+    }
+
+    private User createAndSaveUserFor(Tenant savedTenant) {
+        User user = new User();
+        user.setAuthority(Authority.TENANT_ADMIN);
+        user.setEmail("tenantAdmin@test.com");
+        user.setFirstName("tenantAdmin");
+        user.setLastName("tenantAdmin");
+        user.setTenantId(savedTenant.getId());
+        return userService.saveUser(user);
+    }
+
+    private Tenant createAndSaveTenant(TenantProfile savedProfile) {
+        Tenant tenant = new Tenant();
+        tenant.setTitle("My tenant");
+        tenant.setTenantProfileId(savedProfile.getId());
+        return tenantService.saveTenant(tenant);
+    }
+
+    private TenantProfile createAndSaveTenantProfile() {
+        TenantProfile tenantProfile = new TenantProfile();
+        tenantProfile.setName("Test tenant profile");
+        return tenantProfileService.saveTenantProfile(TenantId.SYS_TENANT_ID, tenantProfile);
+    }
+
+    private OtaPackage createFirmware(TenantId tenantId, DeviceProfileId deviceProfileId) {
```

Comment:
If you found that code somewhere in the project and just copied it here, please, consider using the already existing code instead of copying and pasting. The entity constructor might be changed somehow in the future and this test may be hard to maintain.